agent/azure-monitor-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/MetricDataMapper.java

```diff
@@ -190,11 +198,26 @@ public static void updateMetricPointBuilder(
             metricTelemetryBuilder, statusCode, success, dependencyType, target, isSynthetic);
       }
     } else {
-      attributes.forEach(
-          (key, value) -> metricTelemetryBuilder.addProperty(key.getKey(), value.toString()));
+      setExtraAttributes(metricTelemetryBuilder, attributes);
     }
   }
 
+  private static void setExtraAttributes(
+      AbstractTelemetryBuilder telemetryBuilder, Attributes attributes) {
+    attributes.forEach(
+        (key, value) -> {
+          String stringKey = key.getKey();
+          if (SpanDataMapper.applyConnectionStringAndRoleNameOverrides(
+              telemetryBuilder, value, stringKey)) {
+            return;
+          }
```

Comment:
this check is no longer needed.

Reply:
I think this is needed for custom metrics (so `ai.preview.role_name` can be added as attribute/dimension on custom metrics)

Reply:
i see. it was inside ispreaggregated block and it didn't cover custom metric else block.